lib/assignDisabledRanges.mjs

```diff
@@ -61,101 +65,148 @@ export default function assignDisabledRanges(root, result) {
-		if (
-			!(
-				!isStandardSyntaxComment(comment) &&
-				isConfigurationComment(comment, configurationComment) &&
-				nextComment &&
-				nextComment.type === 'comment' &&
-				(comment.text.includes('--') || nextComment.text.startsWith('--'))
-			)
-		) {
-			checkComment(comment);
+			let lastLine = node.source?.end?.line ?? 0;
+			const fullComment = node.clone();
 
-			return;
-		}
+			let current = nextComment;
 
-		let lastLine = (comment.source && comment.source.end && comment.source.end.line) || 0;
-		const fullComment = comment.clone();
+			while (
+				!isStandardSyntaxComment(current) &&
+				!isConfigurationComment(current.text, configurationComment)
+			) {
+				const currentLine = current.source?.end?.line ?? 0;
 
-		let current = nextComment;
+				if (lastLine + 1 !== currentLine) break;
 
-		while (
-			!isStandardSyntaxComment(current) &&
-			!isConfigurationComment(current, configurationComment)
-		) {
-			const currentLine = (current.source && current.source.end && current.source.end.line) || 0;
+				fullComment.text += `\n${current.text}`;
+
+				if (fullComment.source && current.source) {
+					fullComment.source.end = current.source.end;
+				}
 
-			if (lastLine + 1 !== currentLine) break;
+				inlineEnd = current;
+				const next = current.next();
 
-			fullComment.text += `\n${current.text}`;
+				if (!next || !isComment(next)) break;
 
-			if (fullComment.source && current.source) {
-				fullComment.source.end = current.source.end;
+				current = next;
+				lastLine = currentLine;
 			}
 
-			inlineEnd = current;
-			const next = current.next();
+			checkComment(fullComment, fullComment.source, fullComment.text);
+		}
+
+		if (isRule(node)) {
+			let offset = 0;
+			const selector = node.raws?.selector?.raw;
+
+			checkCommentsInNode(node, selector, offset);
+
+			offset += selector?.length ?? node.selector.length;
+			const between = node.raws?.between;
+
+			checkCommentsInNode(node, between, offset);
+		}
+
+		if (isAtRule(node)) {
+			let offset = node.name.length + 1; // `@` + name
+			const afterName = node.raws?.afterName;
+
+			checkCommentsInNode(node, afterName, offset);
 
-			if (!next || next.type !== 'comment') break;
+			offset += afterName?.length ?? 0;
+			const params = node.raws?.params?.raw;
 
-			current = next;
-			lastLine = currentLine;
+			checkCommentsInNode(node, params, offset);
+
+			offset += params?.length ?? node.params.length;
+			const between = node.raws?.between;
+
+			checkCommentsInNode(node, between, offset);
 		}
 
-		checkComment(fullComment);
+		if (isDeclaration(node)) {
+			let offset = node.prop.length;
+			const between = node.raws?.between;
+
+			checkCommentsInNode(node, between, offset);
+
+			offset += between?.length ?? 0;
+			const value = node.raws?.value?.raw;
+
+			checkCommentsInNode(node, value, offset);
+		}
 	});
 
 	return result;
 
 	/**
-	 * @param {PostcssComment} comment
+	 * @param {PostcssNode} node
+	 * @param {Source | undefined} source
```

Comment:
[suggest] Thanks for refactoring! Furthermore, we can simplify the `source` argument type like this, right?

```suggestion
	 * @param {Source} source
```

If so, using early return could make the code a bit more readable, I think.

```js
if (!source.start) return;

const line = source.start.line;
// ...
```

I think this could be applied to other `processDisable*` functions.

Reply:
good catch 🙇